cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/SparkStreamingPipelineDriver.java

```diff
@@ -80,26 +92,74 @@ public void run(final JavaSparkExecutionContext sec) throws Exception {
     boolean checkpointsDisabled = pipelineSpec.isCheckpointsDisabled();
 
     String checkpointDir = null;
+    JavaSparkContext context = null;
     if (!checkpointsDisabled) {
-      // Get the location of the checkpoint directory.
       String pipelineName = sec.getApplicationSpecification().getName();
-      String relativeCheckpointDir = pipelineSpec.getCheckpointDirectory();
-
-      // there isn't any way to instantiate the fileset except in a TxRunnable, so need to use a reference.
-      final AtomicReference<Location> checkpointBaseRef = new AtomicReference<>();
-      Transactionals.execute(sec, new TxRunnable() {
-        @Override
-        public void run(DatasetContext context) throws Exception {
-          FileSet checkpointFileSet = context.getDataset(DataStreamsApp.CHECKPOINT_FILESET);
-          checkpointBaseRef.set(checkpointFileSet.getBaseLocation());
+      String configCheckpointDir = pipelineSpec.getCheckpointDirectory();
+      if (Strings.isNullOrEmpty(configCheckpointDir)) {
+        // Use the directory of a fileset dataset if the checkpoint directory is not set.
+        Admin admin = sec.getAdmin();
+        if (!admin.datasetExists(DEFAULT_CHECKPOINT_DATASET_NAME)) {
```

Comment:
I guess this means that all Spark pipelines in a namespace share, by default, the same checkpoint directory (this dataset's base path). Will this work in the future in multi-tenant setups with more fine grained access control?

Reply:
We may need to prefix the path in the future with an additional field